runtime/src/lib.rs

```diff
@@ -350,6 +353,79 @@ impl pallet_multisig::Config for Runtime {
     type WeightInfo = weights::pallet_multisig::NHWeight<Runtime>;
 }
 
+parameter_types! {
+    pub const PreimageBaseDeposit: Balance = deposit(2, 64);
+    pub const PreimageByteDeposit: Balance = deposit(0, 1);
+    pub const PreimageHoldReason: RuntimeHoldReason = RuntimeHoldReason::Preimage(pallet_preimage::HoldReason::Preimage);
+}
+
+impl pallet_preimage::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type WeightInfo = weights::pallet_preimage::NHWeight<Runtime>;
+    type Currency = Balances;
+    type ManagerOrigin = EnsureRoot<AccountId>;
+    type Consideration = frame_support::traits::fungible::HoldConsideration<
+        AccountId,
+        Balances,
+        PreimageHoldReason,
+        frame_support::traits::LinearStoragePrice<
+            PreimageBaseDeposit,
+            PreimageByteDeposit,
+            Balance,
+        >,
+    >;
+}
+
+parameter_types! {
+    pub MaximumSchedulerWeight: Weight = BlockWeights::get().max_block;
+    pub MaxScheduledPerBlock: u32 = 50;
+}
+
+pub struct EnsureRootOrSudo<T>(sp_std::marker::PhantomData<T>);
+impl<T, O> EnsureOrigin<O> for EnsureRootOrSudo<T>
+where
+    T: frame_system::Config + pallet_sudo::Config,
+    O: Into<Result<RawOrigin<T::AccountId>, O>> + Clone + From<RawOrigin<T::AccountId>>,
+{
+    type Success = T::AccountId;
+
+    fn try_origin(o: O) -> Result<Self::Success, O> {
+        let origin_result: Result<RawOrigin<T::AccountId>, O> = o.clone().into(); // Clone `o` before converting
+        match origin_result {
+            Ok(RawOrigin::Root) => Ok(pallet_sudo::Pallet::<T>::key().expect("No sudo key set")),
+            Ok(RawOrigin::Signed(who)) => {
+                if Some(who.clone()) == pallet_sudo::Pallet::<T>::key() {
+                    Ok(who)
+                } else {
+                    Err(o)
+                }
+            }
+            _ => Err(o),
+        }
+    }
+
+    #[cfg(feature = "runtime-benchmarks")]
+    fn try_successful_origin() -> Result<O, ()> {
+        pallet_sudo::Pallet::<T>::key()
+            .map(|key| O::from(RawOrigin::Signed(key)))
+            .or_else(|| Some(O::from(RawOrigin::Root))) // Fallback to root if no sudo key
+            .ok_or(())
+    }
+}
+impl pallet_scheduler::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type RuntimeOrigin = RuntimeOrigin;
+    type PalletsOrigin = OriginCaller;
+    type RuntimeCall = RuntimeCall;
+    type MaximumWeight = MaximumSchedulerWeight;
+    type ScheduleOrigin = EnsureRootOrSudo<Runtime>;
+
```

Comment:
I think that we don't need this. `EnsureRoot` is enought.

We can just use a sudo account to schedule the action and that's will forward the ation to the scheduler pallet as it is Root. (I've tryed it)


```suggestion
    type ScheduleOrigin = EnsureRoot<AccountId>;
```